pkg/libdevfile/libdevfile.go

```diff
@@ -119,3 +120,68 @@ func execDevfileEvent(devfileObj parser.DevfileObj, events []string, handler Han
 	}
 	return nil
 }
+
+// GetContainerComponents returns a slice of container components in the given devfile
+func GetContainerComponents(devfileObj parser.DevfileObj) ([]v1alpha2.Component, error) {
+	return devfileObj.Data.GetComponents(common.DevfileOptions{
+		ComponentOptions: common.ComponentOptions{ComponentType: v1alpha2.ContainerComponentType},
+	})
+}
```

Comment:
I wonder if it is really necessary to add this method. It adds nothing, only changing the way to filter on the component type (one method for each type, instead of passing the type as parameter)